src/terragraph-e2e/e2e/minion/BgpUtils.cpp

```diff
@@ -172,67 +165,62 @@ BgpUtils::fetchVtyshBgpStatus() {
   }
 
   std::string bgpSummary = bgpSummaryOutput.value();
-  std::vector<std::string> summaryHeaders, summaryEntries;
-  int neighborCount = parseVtyshBgpTable(
-      bgpSummary, "Neighbor", summaryHeaders, summaryEntries);
 
-  // Iterate through neighbors and create BgpInfo per neighbor
-  for (int i = 0; i < neighborCount; ++i) {
-    // BGP Headers:
-    // Neighbor V AS MsgRcvd MsgSent TblVer InQ OutQ Up/Down State/PfxRcd
+ // dynamic object for  bgpSummary
+ folly::dynamic bgpSummaryInfo = folly::dynamic::object;
+
+ bgpSummaryInfo = folly::parseJson(bgpSummary)["peers"];
 
-    int startRowIdx = i * summaryHeaders.size();
+ // Iterate through neighbors and create BgpInfo per neighbor
+ for (const auto &summary : bgpSummaryInfo.items()) {
+    // Bgp Neighbor Information
     thrift::BgpInfo neighbor;
-    // State/PfxRcd Header
-    auto stateOrPfxToInt = folly::tryTo<int>(summaryEntries[startRowIdx + 9]);
-
-    neighbor.ipv6Address = summaryEntries[startRowIdx]; // Neighbor Header
-    // Need to check if the stateOrPfx is an int since it can either
-    // be a string referring to its state or the actual neighbor prefix
-    neighbor.online = stateOrPfxToInt.hasValue();
-    // AS Header
-    neighbor.asn = folly::to<int>(summaryEntries[startRowIdx + 2]);
-    neighbor.upDownTime = summaryEntries[startRowIdx + 8]; // Up/Down Header
-    // State/PfxRcd Header
-    neighbor.stateOrPfxRcd = summaryEntries[startRowIdx + 9];
+
+    // Neighbor ipv6 Address
+    std::string ipv6Address = summary.first.asString();
+    neighbor.ipv6Address = ipv6Address;
+
+    // online
+    neighbor.online = bgpSummaryInfo[ipv6Address.c_str()]["pfxRcd"].asBool();
```

Comment:
Look up `bgpSummaryInfo[ipv6Address.c_str()]` once instead of repeatedly calling this

Reply:
Addressed the review comments